main.go

```diff
@@ -17,23 +17,38 @@ import (
 
 var cacheDir string
 var listen string
+var gopath string
+var gpEnv string
 
 func init() {
 	flag.StringVar(&listen, "listen", "0.0.0.0:8081", "service listen address")
+	flag.StringVar(&gopath, "gopath", os.Getenv("GOPATH"), "default go path")
 	flag.Parse()
 }
 
 func main() {
-	gpEnv := os.Getenv("GOPATH")
-	if gpEnv == "" {
-		panic("can not find $GOPATH")
+	// setup correct gopath
+	if gopath == "" {
+		gpEnv := os.Getenv("GOPATH")
```

Comment:
I think it's no need to check `GOPATH` env again. The default value is `os.Getenv("GOPATH")`.

```
if gopath == "" {
        panic("can not find $GOPATH")
}else{
...
```